metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/NetworkFunctions.java

```diff
@@ -232,10 +237,15 @@ private static URL toUrl(Object urlObj) {
     if(urlObj == null) {
       return null;
     }
-    try {
-      return new URL(urlObj.toString());
-    } catch (MalformedURLException e) {
-      return null;
+    if(urlObj instanceof String) {
+      try {
+        return new URL(urlObj.toString());
+      } catch (MalformedURLException e) {
```

Comment:
so - do you want to handle InvalidArgumentException for InternetDomainName.from() which throws for invalid domain names?  It should return null as well then?